libs/datalayout/src/test/java/com/linkedin/openhouse/datalayout/generator/OpenHouseDataLayoutGeneratorTest.java

```diff
@@ -0,0 +1,62 @@
+package com.linkedin.openhouse.datalayout.generator;
+
+import com.linkedin.openhouse.datalayout.datasource.TableFileStats;
+import com.linkedin.openhouse.datalayout.datasource.TablePartitionStats;
+import com.linkedin.openhouse.datalayout.strategy.DataLayoutOptimizationStrategy;
+import com.linkedin.openhouse.tablestest.OpenHouseSparkITest;
+import java.util.List;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class OpenHouseDataLayoutGeneratorTest extends OpenHouseSparkITest {
+  @Test
+  void testStrategySanityCheck() throws Exception {
+    final String testTable = "db.test_table_sanity_check";
+    try (SparkSession spark = getSparkSession()) {
+      spark.sql("USE openhouse");
+      spark.sql(
+          String.format(
+              "create table %s (id int, data string, ts timestamp) partitioned by (days(ts))",
+              testTable));
+
+      // produce 2 partitions
+      for (int i = 0; i < 3; ++i) {
+        spark.sql(
+            String.format(
+                "insert into %s values (0, 'data', cast('2024-07-15 00:1%d:34' as timestamp))",
+                testTable, i));
+      }
+      for (int i = 0; i < 3; ++i) {
+        spark.sql(
+            String.format(
+                "insert into %s values (0, 'data', cast('2024-07-16 00:1%d:34' as timestamp))",
+                testTable, i));
+      }
+
+      TableFileStats tableFileStats =
+          TableFileStats.builder().tableName(testTable).spark(spark).build();
+      TablePartitionStats tablePartitionStats =
+          TablePartitionStats.builder().tableName(testTable).spark(spark).build();
+      OpenHouseDataLayoutGenerator strategyGenerator =
+          OpenHouseDataLayoutGenerator.builder()
+              .tableFileStats(tableFileStats)
+              .tablePartitionStats(tablePartitionStats)
+              .build();
+      List<DataLayoutOptimizationStrategy> strategies = strategyGenerator.generate();
+      Assertions.assertEquals(1, strategies.size());
+      DataLayoutOptimizationStrategy strategy = strategies.get(0);
+      // few groups, expect 1 commit
+      Assertions.assertEquals(1, strategy.getConfig().getPartialProgressMaxCommits());
+      Assertions.assertTrue(strategy.getConfig().isPartialProgressEnabled());
+
+      Assertions.assertTrue(
+          strategy.getGain() >= 4,
+          "Gain for 6 files compaction in 2 partitions should be at least 4");
```

Comment:
Shouldn't it be exactly 4 since 3+3 files (3 files each in 2 partitions) would be compacted to 1+1?

Reply:
It would be exactly 4 if we grouped files by partition when calculating cost/gain, it's not happening now due to complexity and low ROI, for production tables I believe the error wouldn't matter.

Reply:
Okay but this is just an assertion for the gain check, right? So for the gain to be correctly calculated, it should be an equivalence check here?

Reply:
Yes, we can make it exact and change the value after taking partitions into account later.